pipeline/risteys_pipeline/finregistry/sample.py

```diff
@@ -0,0 +1,49 @@
+"""Functions for sampling the data"""
+
+import numpy as np
+from random import sample
+from risteys_pipeline.log import logger
+
+
+def sample_cases_and_controls(df, n_cases=250000, controls_per_case=2):
+    """
+    Create a sample of cases and controls and calculate case-cohort weights.
+    
+    Returns:
+        - sample_of_cases (list): finregistryids for cases
+        - sample_of_controls (list): finregistryids for controls 
+        - weight_cases (float): case-cohort weight for cases
+        - weight_controls (float): case-cohort weight for controls
+    """
+    cases = df.loc[df["outcome"] == 1, "finregistryid"].tolist()
```

Comment:
Probably not worth changing, but worth mentioning:

You can also use `df.shape` to get the size of a pandas dataframe, instead of using `df.tolist()` and then `len()` on it.
For example:
```python
cases, _ = df.loc[df["outcome"] == 1, "finregistryid"].shape
```

Reply:
Here `cases` is a list of finregistryids that I used later for sampling and calculating the case-cohort weights. I replaced `cases` with a better variable name `caseids`: 0b3a1ff